kernel/thread.c

```diff
@@ -800,7 +800,7 @@ void z_init_thread_base(struct _thread_base *thread_base, int priority,
 		       uint32_t initial_state, unsigned int options)
 {
 	/* k_q_node is initialized upon first insertion in a list */
```

Comment:
unrelated to the change, what is `k_q_node`? seems to be some old craft

Reply:
It's called "qnode_dlist" or "qnode_rb" now (depending on the scheduler queue backend selected).  This deserves a rework.  Really the whole "thread_base" struct could go away and be folded into thread, the only user of it is the weird dummy thread trick in pipe and mailbox, and they could have their own separate API what they want.